retriever/lib/f_datapackage.py

```diff
@@ -0,0 +1,124 @@
+import os.path
+import yaml
+try:
+    from datapackage import Package 
+except ImportError as e:
+    print("Could not import Package - ",e) 
+from collections import OrderedDict
+from retriever.lib.templates import TEMPLATES
+from retriever.lib.models import myTables
+
+from retriever.lib.defaults import SCRIPT_SEARCH_PATHS
+
+
+"""  Uses url in datapackage.yml to load resources, then downloads the datapackage using datapackage-py
+     https://github.com/frictionlessdata/datapackage-py
+     cleans the json file with the required changes, then creates script module 
+"""
+
+def get_dps():
+    """Load the list of data packages from datapackages.yml
+    Checks all of the search paths for a datapackages.yml file and loads it into
+    a dictionary.
+    """
+    for path in SCRIPT_SEARCH_PATHS:
+        try:
+            with open(os.path.join(path, "datapackages.yml"), 'r') as dp_file:
```

Comment:
The solution to use the first script that you find would be to exit the for loop if you find the first file 

```python
def get_dps():
    """Load the list of data packages from datapackages.yml
    Checks all of the search paths for a datapackages.yml file and loads it into
    a dictionary.
    """
    for path in SCRIPT_SEARCH_PATHS:
        try:
            if os.path.isfile(os.path.join(path, "datapackages.yml")):
                with open(os.path.join(path, "datapackages.yml"), 'r') as dp_file:
                    dp_dict = yaml.safe_load(dp_file)
                return dp_dict
        except:
            pass
    return None
```
With this method, developers will always use the file in the working directory/scripts and users will always use ~/.retriever/scripts  as the defaults.